pkg/gce-cloud-provider/compute/token_source.go

```diff
@@ -0,0 +1,91 @@
+/*
```

Comment:
This seems fine for now but it would be nice to get this into a place where the code can be shared. I realize the one in cloud provider carries too many dependencies. We did the same thing here:

https://github.com/kubernetes/cloud-provider-gcp/blob/master/cmd/gke-exec-auth-plugin/alt_token_source.go

Reply:
Sounds good, if there's a tracking issue for a refactor somewhere I'm happy to link to it

Reply:
If theres no existing issue could you create an issue and link to it, thanks!